common/djangoapps/entitlements/models.py

```diff
@@ -1,8 +1,118 @@
 import uuid as uuid_tools
+from datetime import datetime, timedelta
 
+import pytz
 from django.conf import settings
+from django.contrib.sites.models import Site
 from django.db import models
+
+from certificates.models import GeneratedCertificate  # pylint: disable=import-error
 from model_utils.models import TimeStampedModel
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+class CourseEntitlementPolicy(models.Model):
+    """
+    Represents the Entitlement's policy for expiration, refunds, and regaining a used certificate
+    """
+
+    DEFAULT_EXPIRATION_PERIOD_DAYS = 450
+    DEFAULT_REFUND_PERIOD_DAYS = 60
+    DEFAULT_REGAIN_PERIOD_DAYS = 14
+
+    expiration_period = models.DurationField(
+        default=timedelta(days=DEFAULT_EXPIRATION_PERIOD_DAYS),
+        help_text="Number of days from when an entitlement is created until when it is expired.",
+        null=False
+    )
+    refund_period = models.DurationField(
+        default=timedelta(days=DEFAULT_REFUND_PERIOD_DAYS),
+        help_text="Number of days from when an entitlement is created until when it is no longer refundable",
+        null=False
+    )
+    regain_period = models.DurationField(
+        default=timedelta(days=DEFAULT_REGAIN_PERIOD_DAYS),
+        help_text=("Number of days from when an entitlement is redeemed for a course run until "
+                   "it is no longer able to be regained by a user."),
+        null=False
+    )
+    site = models.ForeignKey(Site)
+
+    def get_days_until_expiration(self, entitlement):
+        """
+        Returns an integer of number of days until the entitlement expires. Includes if the
+        """
+        now = datetime.now(tz=pytz.UTC)
+        expiry_date = entitlement.created + self.expiration_period
+        days_until_expiry = (expiry_date - now).days
+        if not entitlement.enrollment_course_run:
+            return days_until_expiry
+        else:
```

Comment:
Nit: no need for `else`, since if the `if` case hits you will already have returned.